src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.WindowsSettings/Main.cs

```diff
@@ -0,0 +1,319 @@
+            return ContextMenuHelper.GetContextMenu(selectedResult, _assemblyName);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Wrapper method for <see cref="Dispose"/> that dispose additional objects and events form the plugin itself
+        /// </summary>
+        /// <param name="disposing">Indicate that the plugin is disposed</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed || !disposing)
+            {
+                return;
+            }
+
+            if (!(_context is null))
+            {
+                _context.API.ThemeChanged -= OnThemeChanged;
+            }
+
+            _disposed = true;
+        }
+
+        /// <summary>
+        /// Gets the localized name.
+        /// </summary>
+        public string GetTranslatedPluginTitle()
+        {
+            return Name;
+        }
+
+        /// <summary>
+        /// Gets the localized description.
+        /// </summary>
+        public string GetTranslatedPluginDescription()
+        {
+            return Description;
+        }
+
+        /// <summary>
+        /// Change all theme-based elements (typical called when the plugin theme has changed)
+        /// </summary>
+        /// <param name="oldtheme">The old <see cref="Theme"/></param>
+        /// <param name="newTheme">The new <see cref="Theme"/></param>
+        private void OnThemeChanged(Theme oldtheme, Theme newTheme)
+        {
+            UpdateIconPath(newTheme);
+        }
+
+        /// <summary>
+        /// Update all icons (typical called when the plugin theme has changed)
+        /// </summary>
+        /// <param name="theme">The new <see cref="Theme"/> for the icons</param>
+        private void UpdateIconPath(Theme theme)
+        {
+            _defaultIconPath = theme == Theme.Light || theme == Theme.HighContrastWhite
+                ? _lightSymbol
+                : _darkSymbol;
+        }
+
+        /// <summary>
+        /// Translate all setting entires
+        /// </summary>
+        private void TranslateAllSettings()
+        {
+            if (_settingsList is null)
+            {
+                return;
+            }
+
+            foreach (var settings in _settingsList)
+            {
+                var area = Resources.ResourceManager.GetString($"Area{settings.Area}");
+                var name = Resources.ResourceManager.GetString(settings.Name);
+
+                if (string.IsNullOrEmpty(area))
+                {
+                    Log.Warn($"Resource string for [Area{settings.Area}] not found", typeof(Main));
+                }
+
+                if (string.IsNullOrEmpty(name))
+                {
+                    Log.Warn($"Resource string for [{settings.Name}] not found", typeof(Main));
+                }
+
+                settings.Area = area ?? settings.Area;
```

Comment:
Can this crash if json has no string for area in settings element?

(Same on other translations.)

Reply:
I have add a extra null check and use `string.empty` when it is null